TagFilterModule/tag_filter.py

```diff
@@ -0,0 +1,195 @@
+            return tag_name_data
+
+    # returns data with address
+    def locations(self):
+        endpoint = self.api_url + "/api/location/getLocations"
+        payload = {
+        }
+        resp = self.api_sess.post(endpoint, json=payload,
+                       verify=False)
+        order_content = resp.content.decode('utf8')
+        # Load the JSON to a Python list & dump it back out as formatted JSON
+        location_data = json.loads(order_content)
+        return location_data
+
+    # converts lat and lon to street address
+    def location_data_print(self, location_data, lat_1):
+        for value in location_data['locations']:
+            lat_2 = round(value['latitude'], 2)
+            if lat_1 == lat_2:
+                address = value['address1'] + ' ' + value['address2']
+                return address
+
+    # returns data that should be put in CSV file
+    def tag_data(self):   
+        endpoint = self.api_url + "/api/proximity/getLocomotionEventsForTag"
+        self.tag_name_data = self.tag_name()
+        self.uuid = self.tag_name_convert()
+
+        if self.args.startTime:
+            self.start_time = self.milliseconds_time(self.args.startTime)
+        else:
+            self.start_time = int(round((time.time() - (3600 * 24)) * 1000))  # start time defaults to 24 hours ago
+        if self.args.endTime:
+            self.end_time = self.milliseconds_time(self.args.endTime)
+        else:
+            self.end_time = int(round(time.time() * 1000))    # end time defaults to now
+
+        payload = {
+            "movementFilter": self.args.movement,
+            "tagUuid": self.uuid,
+            "createdBeforeMs": self.end_time,
+            "createdAfterMs": self.start_time
+        }
+        resp = self.api_sess.post(endpoint, json=payload,
+                        verify=False)
+        order_content = resp.content.decode('utf8')
+        data = json.loads(order_content)
+        return data
+
+    def execute(self):
+        self.header = ['Tag Name', 'Address', 'Movement', 'Date', 'Movement']
+        self.data = self.tag_data()  
+        self.tag_name_data = self.tag_name()    
+        self.location_data = self.locations()
+        self.count = 0
+        big_list = []
+
+        if 'locomotionEvents' not in self.data:
+            print("Tag name does not exist within organizaton.")
+            return 
+
+        for event in self.data['locomotionEvents']: 
+            small_list = []
+            real_uuid = self.tag_name_convert()
+            if self.args.name and not event['tagUuid'] == real_uuid:
+                continue
+            if self.args.movement and not event['movement'] == self.args.movement:
+                continue
+            # setting up info to add
+            self.real_name = self.tag_uuid_convert(event['tagUuid'], self.tag_name_data)
+            lat_1 = round(event['gpsLocation']['lat'], 2)
+            address = self.location_data_print(self.location_data, lat_1)
```

Comment:
The better way to get the address of the sensor is to use the `baseStationUuid` from the locomotion event and then look up that device and get the location uuid and then grab the location from the locations list.  The base station uuid corresponds to the camera or audio gateway that received the motion beacon (closest one to the tag) and those devices will have a corresponding location.